broker/src/main/java/io/zeebe/broker/logstreams/AtomixLogCompactor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.logstreams;
+
+import io.atomix.protocols.raft.partition.impl.RaftPartitionServer;
+import io.zeebe.broker.Loggers;
+import java.util.concurrent.CompletableFuture;
+
+public final class AtomixLogCompactor implements LogCompactor {
+  private final RaftPartitionServer partitionServer;
+
+  public AtomixLogCompactor(final RaftPartitionServer partitionServer) {
+    this.partitionServer = partitionServer;
+  }
+
+  /**
+   * Sets the compactable index on the Atomix side and triggers compaction. On failure will log the
+   * error but will return a "successful" future - arguable if this is desired behavior.
+   *
+   * @param compactionBound the upper index compaction bound
+   * @return a future which is completed after compaction is finished
+   */
+  @Override
+  public CompletableFuture<Void> compactLog(final long compactionBound) {
+    Loggers.DELETION_SERVICE.debug("Compacting Atomix log up to index {}", compactionBound);
+    partitionServer.setCompactableIndex(compactionBound);
+    return partitionServer.snapshot().exceptionally(error -> onCompaction(compactionBound, error));
+  }
+
+  private Void onCompaction(final long compactionBound, final Throwable error) {
```

Comment:
> As you wrote in the comment I think it make sense to complete the future exceptionally?

I wasn't sure here, since what exactly would we do with it? Right now nothing, except maybe log it, which we already do here. Though it might be more intuitive if here we return a failed future, and the consumer can decide to log it or not, and we don't log it here, I guess.

Reply:
> Though it might be more intuitive if here we return a failed future, and the consumer can decide to log it or not, and we don't log it here, I guess.